### PR TITLE
refactor: move permission migration into executor

### PR DESCRIPTION
**Pull request by mversic**
### Linked issue

Closes #{issue_number} 

### Benefits

we no longer depend on the `ExecutorDataModel::permissions` being correct. 
It's just for users to query it the same way as `ExecutorDataModel::custom_instruction` is

### Checklist

- [ ] I've read `CONTRIBUTING.md`
- [ ] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

**Comment by mversic** [comment]
I can't proceed with this PR atm due to #4743

**Review by Erigara — COMMENTED** [review]
This change looks positive

**Review by dima74 — DISMISSED** [review]

**Comment by dima74** [comment]
Looks like `executor_upgrade_should_run_migration` and `migration_should_cause_upgrade_event` tests are failing

**Comment by mversic** [comment]
> Looks like `executor_upgrade_should_run_migration` and `migration_should_cause_upgrade_event` tests are failing

dang. I'll fix

File: client/tests/integration/smartcontracts/executor_with_custom_permission/src/lib.rs [hunk 1]
```diff
@@ -108,34 +108,15 @@ impl Executor {
...
+    fn replace_token(accounts: &[Account]) -> MigrationResult {
         let can_unregister_domain_definition_id =
             iroha_executor::default::permissions::domain::CanUnregisterDomain::id();
 
         let can_control_domain_lives_definition_id = token::CanControlDomainLives::id();
 
         accounts
             .iter()
-            .try_for_each(|(account, domain_id)| {
-                Revoke::permission(
```
Comment (mversic) [inline client/tests/integration/smartcontracts/executor_with_custom_permission/src/lib.rs:120]: the permission is now revoked on migration

**Review by Erigara — APPROVED** [review]

**Review by dima74 — APPROVED** [review]

**Pull request merged**